src/uct/base/uct_iface.h

```diff
@@ -102,6 +102,18 @@ enum {
     } while (0)
 
 
+/**
+ * Output diagnostic message in terminal and copy it into buffer when _condition is true.
+ */
+#define UCT_OUTPUT_DIAGNOSTIC_MESSAGE(_condition, _buffer, _length, _diag_message, ...) \
```

Comment:
pls replace this macro by a specific function (not macro) for reachability test, it will fill info string and print debug message, something like:
```
uct_iface_unreachable(params, "different addr_family detected. local %s remote %s",
                                       ucs_sockaddr_address_family_str(local_ib_addr_af),
                                       ucs_sockaddr_address_family_str(remote_ib_addr_af));
```

Reply:
@yosefe , I tried to write a function to replace the macro. Just like that:
```
void uct_iface_unreachable(int cond, char* buffer, uint32_t length, const char* fmt, ...)
{
    va_list vars;
    va_start(vars, fmt);
    if (cond) {
        vsnprintf(buffer, length, fmt, vars);
    }
    ucs_debug(fmt, ??????);
    va_end(vars);
}
```

But I hit a problem that ucs_debug is also a macro with a variable number of arguments, if we wanna support function type uct_iface_unreachable, we also should implement another function type "ucs_debug". That makes sense? Thanks.

Reply:
Just like vsnprintf, vucs_debug should be implemented.

Reply:
what about `ucs_debug("%s", buffer);`

Reply:
Sometime the buffer is NULL. In such case, there's no debug trace.